### ISSUE TITLE
[HOLD for payment 2024-07-24] [$250] QBO - Tag status does not change when "Members must tag all expenses" is updated offline

### ISSUE THREAD

**Issue by m-natarajan** [Expensify/App#43754]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___
 
**Version Number:** 1.4.83-1
**Reproducible in staging?:** y
**Reproducible in production?:** n
**If this was caught during regression testing, add the test name, ID and link from TestRail:** n/a
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** applause internal team
**Slack conversation:**

## Action Performed:
Precondition:
- Workspace is connected to QBO.

1. Go to staging.new.expensify.com
2. Go to workspace settings > Tags.
3. Click Settings.
4. Toggle on "Members must tag all expenses".
5. Close the RHP.
6. Note that "Classes" and "Customers/Projects" are labelled with "Required" when "Members must tag all expenses" is enabled and vice versa.
7. Go offline.
8. Click Settings.
9. Toggle off/on "Members must tag all expenses".
## Expected Result:
In offline mode, "Classes" and "Customers/Projects" status will change when "Members must tag all expenses" is changed.

## Actual Result:
In offline mode, "Classes" and "Customers/Projects" status do not change when "Members must tag all expenses" is changed. It only works online.
## Workaround:
unknown
## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android: Native
- [x] Android: mWeb Chrome
- [x] iOS: Native
- [x] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [x] MacOS: Desktop

## Screenshots/Videos

https://github.com/Expensify/App/assets/38435837/30b77958-2074-40ac-a638-b20519ae1f14

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~011dbf2d7b0bb7331f</li>
        <li>Upwork Job ID: 1801613859295852710</li>
        <li>Last Price Increase: 2024-07-03</li>
<li>Automatic offers: </li>
<ul>
<li>Krishna2323 | Contributor | 102994557</li>
</ul></li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @stephanieelliott</details>

**Comment by m-natarajan** [comment]
@laurenreidexpensify FYI I haven't added the External label as I wasn't 100% sure about this issue. Please take a look and add the label if you agree it's a bug and can be handled by external contributors

**Comment by m-natarajan** [comment]
We think that this bug might be related to #wave-collect - Release 1

**Comment by mountiny** [comment]
Marking this as external and not a blocker, this is a rare flow and only offline case. Looking for proposals

**Comment by truph01** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
- In offline mode, "Classes" and "Customers/Projects" status do not change when "Members must tag all expenses" is changed. It only works online.

### What is the root cause of that problem?
- The requirement for Members must tag all expenses is governed by the ```policy.requiresTag property```, which indicates whether users must specify a tag when creating an expense.

- For single-level tags, if ```policy.requiresTag``` is true, it mandates that a single tag must be set when creating an expense.

- For multiple-level tags, let's consider two tag lists, ```State``` and ```Region```. If ```policy.requiresTag``` is ```true```, it traditionally meant that setting either ```State``` or ```Region``` sufficed. However, the current behavior mandates setting both ```State``` and ```Region``` simultaneously when ```policy.requiresTag``` is ```true```, which contradicts the expected logic.

### What changes do you think we should make in order to solve the problem?
- With that, I think we should remove option ```Members must tag all expenses``` in case of multiple level tags. We already do it in OD, where we only have ```Members must tag all expenses``` in case of single level tag.

- If we want to specify which tag list should be required, currently, we can do that by clicking on the tag list then toggle the require option. 

- So only think we need to do is hide this option:
https://github.com/Expensify/App/blob/75c104bfe009ef937bcd129a35bb5a2e52f4339d/src/pages/workspace/tags/WorkspaceTagsSettingsPage.tsx#L64-L72 when it is ```isMultiLevelTags```.
or we can hide the setting button in ```WorkspaceTagsPage``` and prevent user from opening ```WorkspaceTagsSettingsPage``` as there is no option to display here in case of ```isMultiLevelTags```

### What alternative solutions did you explore? (Optional)

**Comment by truph01** [comment]
@mountiny 

- In staging.expensify.com, we do not have the ```Members must tag all expenses``` option in case of multiple level tags, so should we do like that in staging.new.expensify.com?  Because there is a lot of weird behavior when we allow option ```Members must tag all expenses```.

- For example, when the ```Members must tag all expenses``` or ```policy.requiresTag``` is ```true```, there are two possible values of the policyTags:
```
{
  "Region": { "name": "Region", "required": true },
  "State": { "name": "State", "required": false }
}
```
and 
```
{
  "Region": { "name": "Region", "required": true },
  "State": { "name": "State", "required": true }
}
```

**Comment by laurenreidexpensify** [comment]
@Ollyws couple of proposals to review above when you get a chance thanks

**Comment by laurenreidexpensify** [comment]
@Ollyws reasonable suggestion - increased!

**Comment by laurenreidexpensify** [comment]
@Ollyws reasonable suggestion - increased!

**Comment by laurenreidexpensify** [comment]
hey @stephanieelliott handing this one over for my parental leave K2 clear out. we're still waiting to review / confirm proposals on this one. thanks!

**Comment by mountiny** [comment]
Thanks @Ollyws

**Comment by mountiny** [comment]
@Ollyws still no good proposals?

**Comment by truph01** [comment]
IMO, I think this bug can be reproduced with policy that have multiple levels tags (setup in OD) as well, no need QBO.

**Comment by mountiny** [comment]
@Ollyws Could you summarize the problem and post in Slack please? This might be better discussed there. Thank you

**Comment by stephanieelliott** [comment]
Discussion going on in Slack: https://expensify.slack.com/archives/C02NK2DQWUX/p1719490486187669?thread_ts=1714509698.711859&cid=C02NK2DQWUX

**Comment by mountiny** [comment]
Discussing in the Slack thread

**Comment by mountiny** [comment]
Yeah I think we can add this regression test to QA

**Comment by JmillsExpensify** [comment]
@stephanieelliott can I get a payment summary when you get a chance?

**Comment by stephanieelliott** [comment]
Yep, here it is @JmillsExpensify! 

Summarizing payment on this issue: 

 - Contributor: @Krishna2323 $250 via Upwork - Paid via Upwork 
 - Contributor+: @Ollyws $250 via ND  - Requested in ND 
 
 Upwork job is here:  https://www.upwork.com/jobs/~011dbf2d7b0bb7331f

**Comment by stephanieelliott** [comment]
Issue for QA test is here: https://github.com/Expensify/Expensify/issues/new

**Comment by JmillsExpensify** [comment]
$250 approved for @Ollyws
